integration/simulation/output_stats.go

```diff
@@ -39,18 +41,37 @@ func NewOutputStats(simulation *Simulation) *OutputStats {
 }
 
 func (o *OutputStats) populateHeights() {
-	obscuroClient := o.simulation.RPCHandles.ObscuroClients[0]
-	o.l1Height = int(getHeadBlockHeight(obscuroClient))
+	l2Client := o.simulation.RPCHandles.ObscuroClients[0]
+	obscuroClient := obsclient.NewObsClient(l2Client)
+
+	l1Height, err := obscuroClient.BlockNumber()
+	if err != nil {
+		panic(fmt.Errorf("simulation failed because could not read L1 height. Cause: %w", err))
+	}
+	o.l1Height = int(l1Height)
+
 	o.l2Height = int(getHeadRollupHeader(obscuroClient).Number.Uint64())
 }
 
 func (o *OutputStats) countBlockChain() {
 	l1Node := o.simulation.RPCHandles.EthClients[0]
 	l2Client := o.simulation.RPCHandles.ObscuroClients[0]
+	obscuroClient := obsclient.NewObsClient(l2Client)
 
 	// iterate the Node Headers and get the rollups
-	for header := getHeadRollupHeader(l2Client); header != nil && !bytes.Equal(header.Hash().Bytes(), common.GenesisHash.Bytes()); header = getRollupHeader(l2Client, header.ParentHash) {
+	header := getHeadRollupHeader(obscuroClient)
+	var err error
+	for {
```

Comment:
Could maybe keep it in a `while` format and tidy away that break check into a readable method
```
for isNonGenesisHeader(header) {
...
}
```

Reply:
yeah fair, guess it would look like this which isn't really a significant improvement:
```
for nonGenesisHeader(header) {
  o.l2RollupCountInHeaders++
  header, err = nextHeader(...)
  if err != nil {
    ...
  }
}
```